### PR TITLE
Ignore deleted needles when showing needle editor

### PR DESCRIPTION
**Pull request by Martchus**
to fix https://progress.opensuse.org/issues/13194

### CODE REVIEW

**Comment by Martchus** [comment]
Warnings about deleted/bad needles are now shown.
I wanted to use flash messages in the first place, so I also added an improvement for displaying this message. If this is not wanted, just discard it.

**Comment by okurz** [comment]
LGTM

**Comment by okurz** [comment]
@aaannz ?

**Comment by aaannz** [comment]
:+1:

File: templates/step/edit.html.ep [hunk 1]
```diff
@@ -14,6 +14,12 @@
     setupResultButtons();
 % end
 
+% if(@$error_messages) {
+    <div class="alert-warning alert fade in" id="editor_warnings"><button class="close" data-dismiss="alert">x</button>
+        <span><%== '<ul><li>'.join('</li><li>', @$error_messages).'</li></ul>' %></span>
```
Comment (coolo) [inline templates/step/edit.html.ep]: what is %== ?

Comment (Martchus) [inline templates/step/edit.html.ep]: Error messages might contain HTML syntax to allow formatting. This means of course that we must be careful when generating the messages. Currently only $needlename, $distribution and $dversion go into the messages which should be safe without further escaping? If not I can change it when I'm at the office again.

**Pull request merged**